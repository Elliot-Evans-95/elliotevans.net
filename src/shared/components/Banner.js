// @flow

import React from 'react';
import PropTypes from 'prop-types';
import {
  Header,
  HeaderGroup,
  MyFace,
  SubHeading,
  Title
} from '../styles/styles';

const Banner = props => {
  return (
    <Header>
      <MyFace
        src="https://avatars0.githubusercontent.com/u/22132230?s=400&v=4"
        alt="My face"
      />
      <HeaderGroup>
        <Title>{props.text[0]}</Title>
        <SubHeading>{props.text[1]}</SubHeading>
      </HeaderGroup>
    </Header>
  );
};

Banner.propTypes = {
  text: PropTypes.array
};

export default Banner;
