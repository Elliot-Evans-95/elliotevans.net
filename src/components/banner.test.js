import React from 'react';
import { mount, shallow } from 'enzyme';

import Banner from './banner';
import {
  HeaderGroup,
  MyFace,
  SubHeading,
  Title
} from '../assets/styles/styles';

describe('<Banner />', () => {
  const text = [
    'Hello, this is Elliot Evans.',
    'Front-end Developer here...',
    'Example',
    'There would be a short description from one of my blog posts but instead here is some placeholder text'
  ];

  test('render a label', () => {
    const wrapper = shallow(<Banner>Hello Jest!</Banner>);

    expect(wrapper).toMatchSnapshot();
  });

  it('allows us to set props', () => {
    const wrapper = mount(<Banner text={text} />);
    const fistLineInText = wrapper.prop('text')[0];

    expect(fistLineInText).toEqual('Hello, this is Elliot Evans.');
  });

  it('renders <MyFace /> component', () => {
    const wrapper = shallow(<Banner />);
    expect(wrapper.find(MyFace)).toHaveLength(1);
  });

  it('renders <HeaderGroup /> component', () => {
    const wrapper = mount(<HeaderGroup />);
    expect(wrapper.find(HeaderGroup)).toHaveLength(1);
  });

  it('renders <Title /> component', () => {
    const wrapper = mount(<Title />);
    expect(wrapper.find(Title)).toHaveLength(1);
  });

  it('renders <SubHeading /> component', () => {
    const wrapper = mount(<SubHeading />);
    expect(wrapper.find(SubHeading)).toHaveLength(1);
  });
});
