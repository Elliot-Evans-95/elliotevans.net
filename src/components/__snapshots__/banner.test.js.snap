// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> render a label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Banner
    text={
        Array [
            "Hello, this is Elliot Evans.",
            "Front-end Developer here...",
            "Example",
            "There would be a short description from one of my blog posts but instead here is some placeholder text",
          ]
    }
>
    Hello Jest!
</Banner>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.img
          alt="My face"
          src="https://avatars0.githubusercontent.com/u/22132230?s=400&v=4"
/>,
        <styled.div>
          <styled.h1>
                    Hello, this is Elliot Evans.
          </styled.h1>
          <styled.h3>
                    Front-end Developer here...
          </styled.h3>
</styled.div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "alt": "My face",
          "src": "https://avatars0.githubusercontent.com/u/22132230?s=400&v=4",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.h1>
              Hello, this is Elliot Evans.
</styled.h1>,
            <styled.h3>
              Front-end Developer here...
</styled.h3>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Hello, this is Elliot Evans.",
            },
            "ref": null,
            "rendered": "Hello, this is Elliot Evans.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Front-end Developer here...",
            },
            "ref": null,
            "rendered": "Front-end Developer here...",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.img
            alt="My face"
            src="https://avatars0.githubusercontent.com/u/22132230?s=400&v=4"
/>,
          <styled.div>
            <styled.h1>
                        Hello, this is Elliot Evans.
            </styled.h1>
            <styled.h3>
                        Front-end Developer here...
            </styled.h3>
</styled.div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "alt": "My face",
            "src": "https://avatars0.githubusercontent.com/u/22132230?s=400&v=4",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <styled.h1>
                Hello, this is Elliot Evans.
</styled.h1>,
              <styled.h3>
                Front-end Developer here...
</styled.h3>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Hello, this is Elliot Evans.",
              },
              "ref": null,
              "rendered": "Hello, this is Elliot Evans.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Front-end Developer here...",
              },
              "ref": null,
              "rendered": "Front-end Developer here...",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
